test/broadcast.jl

```diff
@@ -213,4 +213,21 @@ end
         foo493(X) = normalize.(X)
         @test foo493(X) isa Core.Compiler.return_type(foo493, Tuple{typeof(X)})
     end
+
+    @testset "broadcasting with tuples" begin
+        # issue 485
+        @test @inferred(SA[1,2,3] .+ (1,))               === SA{Int}[2, 3, 4]
+        @test @inferred(SA[1,2,3] .+ (10, 20, 30))       === SA{Int}[11, 22, 33]
+        @test @inferred((1,2)     .+ (SA[10 20; 30 40])) === SA{Int}[11 21; 32 42]
+        @test @inferred((SA[10 20; 30 40]) .+ (1,2))     === SA{Int}[11 21; 32 42]
+
+        add_bc!(m, v) = m .+= v  # Helper function; @inferred gets confused by .+= syntax
+        @test @inferred(add_bc!(MVector((1,2,3)), (10, 20, 30)))   ::MVector{3,Int}   == SA[11, 22, 33]
+        @test @inferred(add_bc!(MMatrix(SA[10 20; 30 40]), (1,2))) ::MMatrix{2,2,Int} == SA[11 21; 32 42]
+
+        # Tuples of SA
+        @test SA[1,2,3] .* (SA[1,0],) === SVector{3,SVector{2,Int}}(((1,0), (2,0), (3,0)))
+        # Unfortunately this case of nested broadcasting is not inferred
+        @test_broken @inferred(SA[1,2,3] .* (SA[1,0],))
```

Comment:
BTW, this case wasn't inferred before either, so we haven't made it worse in this PR.

Reply:
This inference failure is weird. `Base.copy(B::Broadcasted{StaticArrayStyle{M}})` fails to inline and then its return type is not inferred. But `Cthulhu.jl` says everything's OK when I descend to that method of `copy`. There seems to be a problem outside `StaticArrays`.